platform/ios/MapboxGL.m

```diff
@@ -0,0 +1,48 @@
+#import <Foundation/Foundation.h>
+
+#import "MapboxGL.h"
+#import "NSProcessInfo+MGLAdditions.h"
+
+@interface MapboxGL()
+
+@property (atomic) NSString *accessToken;
+
+@end
+
+
+@implementation MapboxGL
+
+static MapboxGL *_sharedManager;
+
+// Can be called from any thread. Called implicitly from any
+// public class convenience methods.
+//
++ (id) sharedInstanceWithAccessToken:(NSString *)token {
+    static dispatch_once_t onceToken;
+    dispatch_once(&onceToken, ^{
+        if ( ! NSProcessInfo.processInfo.mgl_isInterfaceBuilderDesignablesAgent) {
+            void (^setupBlock)() = ^{
+                _sharedManager = [[self alloc] init];
+                _sharedManager.accessToken = token;
+            };
+            if ( ! [[NSThread currentThread] isMainThread]) {
+                dispatch_sync(dispatch_get_main_queue(), ^{
+                    setupBlock();
+                });
+            }
+            else {
+                setupBlock();
+            }
+        }
+    });
+    return _sharedManager;
+}
+
++ (NSString *) getAccessToken {
```

Comment:
On second thought, if the developer can just get the access token from a class method (`+accessToken`), why make them create a shared instance in the first place?

Let’s strive for parallelism in one of two ways:

 - A class method `+sharedManager` that returns an [`MGLAccountManager`](https://github.com/mapbox/mapbox-gl-native/pull/1329/files#r29024059) with nothing set, and two instance methods: `-setAccessToken:` and `-accessToken`. This would be similar to how `NSProcessInfo` works.
 - Two standalone C functions: `MGLRegisterAccessToken()` and `MGLGetRegisteredAccessToken()`. (On Apple platforms, Objective-C getters don’t say “get”, but C getters do, [to distinguish from “copy” functions](https://developer.apple.com/library/ios/documentation/CoreFoundation/Conceptual/CFMemoryMgmt/Concepts/Ownership.html#//apple_ref/doc/uid/20001148-SW3).) Instead of keeping a singleton object around, keep just a static access token string. This approach is simpler but less extensible.